### PR TITLE
Completely rewrites the orbit screen

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
I set about refactoring Orbit.js when I just decided to scrap it and maybe add in some different functionality.

This PR:
- rewrites orbit from the ground up in typescript
- color coded list + scrollable + wider
- adds some styling and effects to gauge "threat": icons change, buttons change color
- fixes button text overflow
- fixes issue where scrolling hides the search header
- fixes similar antags being grouped separately (nukie and nukie leader, etc)

<details>
<summary>See previous issues</summary>

Screenshot 2022-06-29 153753

</details>

<details>
<summary>Preview</summary>

you can't see it here, but when I type "a" and get multiple results, pressing ENTER will choose the most orbited (or first) result
OfyMa0ILym

heatmap in motion
Screenshot 2022-06-30 162847

</details>

## Why It's Good For The Game
one of the more important UIs is made to be easier to use and better to look at

## Changelog

:cl:
refactor: Orbit UI has been reworked.
/:cl:

### CODE REVIEW

**Comment by calliophis-maculiceps** [comment]
could be made smaller/less clunkier

**Comment by jlsnow301** [comment]
> ultimately I don't like the base of this and don't think it is better.

I've reworked it a bit to feel more like the original, let me know what you think.

**Comment by Mothblocks** [comment]
I like the colors (though I'm not sure orbiting should change the color? I see that mentioned somewhere in the code), though not sure about the gap in the name.

That being said, bigger improvement :+1:

**Comment by jlsnow301** [comment]
> I like the colors (though I'm not sure orbiting should change the color? I see that mentioned somewhere in the code), though not sure about the gap in the name.

It currently only highlights people with large followings, trying to gameify the screen a bit

**Comment by stylemistake** [comment]
Left space on section headers looks unnecessary, adds a visual alignment line that is separate from the buttons, should just be aligned normally.

**Comment by jlsnow301** [comment]
> Left space on section headers looks unnecessary, adds a visual alignment line that is separate from the buttons, which convolutes the layout, it should just be aligned normally.

Done!

**Comment by Mothblocks** [comment]
> It currently only highlights people with large followings, trying to gameify the screen a bit

Can you send a picture of how that looks because in my head I think it would just look weird

**Comment by jlsnow301** [comment]
> Can you send a picture of how that looks because in my head I think it would just look weird

gjTjw3STWF

**Comment by Mothblocks** [comment]
Yeah I don't think I like that honestly, I think it makes it look noisy

CC @stylemistake if you have any interest bikeshedding on it

**Comment by jlsnow301** [comment]
> Yeah I don't think I like that honestly, I think it makes it look noisy

I'm not partial to this "action heatmap" (it's just something I was toying around with), so no offense taken, but I do disagree that it creates noise. 

In average games, there are MANY observable buttons going on and having the 1-2 interesting ones colorized sort of defuses that noise

**Comment by optimumtact** [comment]
seems fine to me?

File: tgui/packages/tgui/interfaces/Orbit.tsx [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+    const { antag } = antagonist;
+    const resolvedName = ANTAG_GROUPS[antag!] || antag;
+    if (collatedAntagonists[resolvedName] === undefined) {
+      collatedAntagonists[resolvedName] = [];
+    }
+    collatedAntagonists[resolvedName].push(antagonist);
+  }
+  const sortedAntagonists: [string, Observable[]][] =
+    Object.entries(collatedAntagonists);
+  sortedAntagonists.sort((a, b) => {
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/Orbit.tsx:362]: Use sortBy instead it makes this code simpler

File: tgui/packages/tgui/interfaces/Orbit.tsx [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+        ?.toLowerCase()
+        ?.includes(searchQuery?.toLowerCase());
+    });
+    filteredLists.push(filtered);
+  });
+  return filteredLists;
+};
+
+/** Sorts by the highest orbiter count */
+const sortByOrbiters = (a: Observable, b: Observable): number => {
```
Comment (Mothblocks) [inline tgui/packages/tgui/interfaces/Orbit.tsx:319]: Use sortBy instead and just a basic key

Comment (jlsnow301) [reply tgui/packages/tgui/interfaces/Orbit.tsx:372]: I ended up using sortBy elsewhere where i had guaranteed props (antagonist.antag), but this had trouble where observable.observers is sometimes undefined. 
I reckon the old fashion sort is much more concise and readable in this case

Also, toying with sortBy has clued me into how the common packages (flow, map, filter) work, and I'm super thankful for that

Comment (jlsnow301) [reply tgui/packages/tgui/interfaces/Orbit.tsx]: nvm sortby superior

**Comment by Zemurr** [comment]
cool now make ashwalkers green in that menu bc they are not real antags

**Comment by jlsnow301** [comment]
> cool now make ashwalkers green in that menu bc they are not real antags

done! cool suggestion. i made them lizard green actually, not normal green
Screenshot 2022-07-01 152431

**Comment by Jamarkus127** [comment]
Why not make it so that every menu is collapsible? The only menus I've seen that I cant collapse are the alive and antag ones. If you are gonna have some of them collapsible, then why not make em all? it looks neater on open too.

**Comment by jlsnow301** [comment]
> Why not make it so that every menu is collapsible? The only menus I've seen that I cant collapse are the alive and antag ones. If you are gonna have some of them collapsible, then why not make em all? it looks neater on orbit menu open too.

i am not opposed to this, but I am trying to keep in line with the previous comments by mothblocks

**Comment by SethLafuente** [comment]
Xenos should have gotten their own purple tab missed opportunity

**Pull request merged**